frontend/src/Components/ContentComponent.tsx

```diff
@@ -400,6 +418,63 @@ export default function ContentComponent(props: ContentComponentProps) {
                  style={{maxHeight: cut && props.autoCut ? props.autoCut: undefined}}
                  dangerouslySetInnerHTML={{__html: props.content}} ref={contentDiv} />
             {cut && <div className={styles.cutCover}><button className={styles.cutButton} onClick={handleCut}>Читать дальше</button></div>}
+            {zoomedImg &&  <ZoomComponent {...zoomedImg} onExit={() => setZoomedImg(null)} />}
         </>
     );
 }
+
+// extract zoom component
+
+interface ZoomComponentProps {
+    src: string;
+    width: number;
+    height: number;
+    onExit: () => void;
+}
+
+function ZoomComponent(props: ZoomComponentProps) {
+    // need to account for retina displays
+    const minScale = Math.min(1, window.innerWidth / props.width / 2, window.innerHeight / props.height / 2);
```

Comment:
What sense to allow scaling to less than screen? 
```suggestion
    const minScale = Math.min(1, window.innerWidth / props.width, window.innerHeight / props.height);
```

Reply:
I think it's reasonable to allow to downscale at least to the original size of the image, if it's less than full screen. However, it's true that I should use here the scale for the original size specifically.